packages/vite/src/node/optimizer/esbuildDepPlugin.ts

```diff
@@ -112,7 +112,8 @@ export function esbuildDepPlugin(
         namespace: 'optional-peer-dep',
       }
     }
-    if (ssr && isBuiltin(resolved)) {
+    // TODO: Should this be environment.config.nodeCompatible or a more fine-grained option?
+    if (environment.config.consumer === 'server' && isBuiltin(resolved)) {
       return
     }
```

Comment:
This is a good idea. Only returning the id from the previous `createResolver` was always a pain we couldn't get out of because it was a public API. I don't know why I follow the same design. We should change this before merging.

Reply:
I started doing this one but it triggers several changes that are unrelated to the PR. I think we could merge the PR first, and then explore this change. It would be easier to check all the places where we will end up with more information to use.